File: kattis/alicedigital.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int testcase;
    cin >> testcase;
    for (int i = 0; i < testcase; i++) {
        int n, m;
        cin >> n >> m;
        int pre = 0, aft = 0;
        int ctr = 0;
        int maxi = -1e7;
        int catetawal;
        int catetakhir;
        bool verify = true;
        int saver;
        for (int i = 0; i < n; i++) {
            int x;
            cin >> x;
            if (x < m) {
                maxi = max(pre, maxi);
                pre = 0;
                aft = 0;
                ctr = 0;
            }
            else {
                if (i == 0) {
                    catetawal = x;
                }
                if (i == n-1) {
                catetakhir = x;
                }
                if (m == x) {
                    if (i == 0) {
                        pre = x;
                        aft = x;
                        ctr++;
                    }
                    else if (ctr == 0) {
                        pre += x;
                        aft = x;
                        maxi = max(pre, maxi);
                        ctr++;
                    }
                    else {
                        maxi = max(maxi, pre);
                        pre = aft;
                        aft = x;
                        ctr++;
                    }
                }
                else {
                    pre += x;
                    aft += x;
                }
            }
        }
        if (ctr == 0) {
        	cout << maxi << endl;
        }
        if (ctr == 1) {
            if (catetawal == m) {
                int a;
                a = max(pre,aft);
                maxi = max(maxi, a);
                cout << maxi - catetakhir << endl; 
            }
            else if (catetakhir == m) {
                int a;
                a = max(pre,aft);
                maxi = max(maxi, a);
                cout << maxi - catetawal << endl;
            }
            else {
                int z, a;
                z = min(catetawal, catetakhir);
                a = max(pre, aft);
                maxi = max(maxi, a);
                cout << maxi-z << endl;
            }
        }
        else {
            int b;
            b = max(pre, aft);
            maxi = max(maxi, b);
            cout << maxi << endl;
        }
    }
}

File: kattis/alicedigital.py
def main():
    testcase = input()
    for i in range (int(testcase)):
        length, m = [int(x) for x in input().split()]
        lst = [int(x) for x in input().split()]
        counter = 0
        pre = 0
        aft = 0
        maxi = -10**9
        verifier = False
        for i in range (len(lst)):
            if lst[i] < m:
                if verifier:
                    maxi = max(maxi, pre) 
                pre = 0
                aft = 0
                verifier = False
                # print("ini pre2")
                # print(pre)
                # print("ini aft2")
                # print(aft)
                # print("ini maxi skrg")
                # print(maxi)
                # print("ini verifier")
                # print(verifier)
            if lst[i] == m:
                if verifier == False:
                    verifier = True
                    pre += lst[i]
                    aft = lst[i]
                    # print("ini pre3")
                    # print(pre)
                    # print("ini aft3")
                    # print(aft)
                    # print("ini maxi skrg")
                    # print(maxi)
                    # print("ini verifier")
                    # print(verifier)
                else:
                    maxi = max(maxi, pre)
                    pre = aft
                    aft = lst[i]
                    # print("ini pre4")
                    # print(pre)
                    # print("ini aft4")
                    # print(aft)
                    # print("ini maxi skrg")
                    # print(maxi)
                    # print("ini verifier")
                    # print(verifier)
            if lst[i] > m:
                pre += lst[i]
                aft += lst[i]
                # print("ini pre5")
                # print(pre)
                # print("ini aft5")
                # print(aft)
                # print("ini maxi skrg")
                # print(maxi)
                # print("ini verifier")
                # print(verifier)
        if verifier:
            maxi = max(maxi, pre)
        print(maxi)






if __name__ == '__main__':
    main()
